apps/base-docs/docs/pages/identity/smart-wallet/guides/paymasters.mdx

```diff
@@ -212,85 +223,56 @@ export const myNFTABI = [
 ] as const;
 
 export const myNFTAddress = "0x119Ea671030FBf79AB93b436D2E20af6ea469a19";
-
 ```
 
 ```ts twoslash [config.ts] filename="config.ts"
-import { createPublicClient, http } from "viem";
+import { createClient, createPublicClient, http } from "viem";
 import { baseSepolia } from "viem/chains";
+import { entryPoint06Address, createPaymasterClient, createBundlerClient } from "viem/account-abstraction";
 
 export const client = createPublicClient({
   chain: baseSepolia,
   transport: http(),
 });
-```
-
-:::
 
-#### Create Proxy
-
-The proxy you create will need to handle the `pm_getPaymasterStubData` and `pm_getPaymasterData` JSON-RPC requests specified by ERC-7677.
-:::code-group
-
-```ts twoslash [route.ts]
-import { paymasterClient } from "./config";
-// @noErrors
-import { willSponsor } from "./utils";
-
-export async function POST(r: Request) {
-  const req = await r.json();
-  const method = req.method;
-  const [userOp, entrypoint, chainId] = req.params;
-  const sponsorable = await willSponsor({ chainId, entrypoint, userOp });
-  if (!sponsorable) {
-    return Response.json({ error: "Not a sponsorable operation" });
-  }
+const paymasterService = process.env.PAYMASTER_SERVICE_URL!;
 
-  if (method === "pm_getPaymasterStubData") {
-    const result = await paymasterClient.getPaymasterStubData({
-      userOperation: userOp,
-    });
-    return Response.json({ result });
-  } else if (method === "pm_getPaymasterData") {
-    const result = await paymasterClient.getPaymasterData({
-      userOperation: userOp,
-    });
-    return Response.json({ result });
-  }
-  return Response.json({ error: "Method not found" });
-}
+export const paymasterClient = createPaymasterClient({
+  transport: http(paymasterService),
+});
 
+export const bundlerClient = createBundlerClient({
+  chain: baseSepolia,
+  paymaster: paymasterClient, 
+  transport: http(paymasterService),
+})
 ```
+:::
 
-```ts twoslash [config.ts] filename="config.ts"
-// @errors: 2305 2554
-// @noErrors
-const process = { env: { PAYMASTER_SERVICE_URL: '' } }
-// ---cut---
-import { createClient, http } from "viem";
-import { baseSepolia } from "viem/chains";
-import { ENTRYPOINT_ADDRESS_V06 } from "permissionless";
-import { paymasterActionsEip7677 } from "permissionless/experimental";
+<Callout type="warning">
+**Create a Proxy**
 
-const paymasterService = process.env.PAYMASTER_SERVICE_URL!;
+As you can see in the Paymaster transaction [component](https://github.com/wilsoncusack/wagmi-scw/blob/main/src/components/TransactWithPaymaster.tsx),
+we use a proxy to protect the paymaster service URL, because it is exposed on the client side.
```

Comment:
Wording could be better : "we use a proxy to protect the paymaster service URL to prevent it from being exposed/leaked on a frontend client"